### PR TITLE
Prove simple iterative sum

### PR DESCRIPTION
**Pull request by yilongli**

### CODE REVIEW

**Comment by yilongli** [comment]
@dwightguth @andreistefanescu please review

File: samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/tests/config_verification.xml [hunk 1]
```diff
@@ -3,14 +3,24 @@
...
         >
     <kompile-option name="--backend" value="java" />
     <all-programs>
-      <krun-option name="--prove" value="verification/reverse_spec.k" />
+      <krun-option name="--prove" value="verification/reverse/reverse_spec.k" />
     </all-programs>
   </test>
-</tests>
 
+  <test definition="simple-untyped.k"
```
Comment (andreistefanescu) [inline samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/tests/config_verification.xml]: I think this is compiling the definition twice. @dwightguth how should this config file look like?

File: src/javasources/KTool/src/org/kframework/backend/java/symbolic/SymbolicRewriter.java [hunk 2]
```diff
@@ -643,6 +643,9 @@ private ConstrainedTerm applyRule(ConstrainedTerm constrainedTerm, List<Rule> ru
         queue.add(initialTerm);
         boolean guarded = false;
         while (!queue.isEmpty()) {
+            if (queue.size() == 1) {
```
Comment (andreistefanescu) [inline src/javasources/KTool/src/org/kframework/backend/java/symbolic/SymbolicRewriter.java]: debug statement?

File: samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/verification/sum_iterative/sum_spec.k [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    </in>
+    <out>
+        _:List
+    </out>
+    <nextLoc>
+        N1:Int => ?N2:Int
+    </nextLoc>
+</T>
+requires X >=Int 0
+ensures ?Y ==Int (X *Int (X +Int 1)) /Int 2
```
Comment (andreistefanescu) [inline samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/verification/sum_iterative/sum_spec.k]: you can manually substitute `?Y` by `(X *Int (X +Int 1)) /Int 2`

File: samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/verification/sum_iterative/sum_spec.k [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    </in>
+    <out>
+        _:List
+    </out>
+    <nextLoc>
+        N1:Int => ?N2:Int
+    </nextLoc>
+</T>
+requires X1 >=Int 0
+ensures ?Y2 ==Int Y1 +Int (X1 *Int (X1 +Int 1)) /Int 2
```
Comment (andreistefanescu) [inline samples/java_rewrite_engine/tutorial/2_languages/1_simple/1_untyped/verification/sum_iterative/sum_spec.k]: see above

**Comment by andreistefanescu** [comment]
Awesome, @yilongli

**Comment by dwightguth** [comment]
@andreistefanescu merge if you are happy.

**Pull request merged**